torch/autograd/__init__.py

```diff
@@ -38,7 +38,46 @@ def backward(variables, grad_variables, retain_variables=False):
             specify ``True`` if you want to differentiate some subgraph multiple
             times.
     """
+    grad_variables = tuple(var if isinstance(var, Variable) or var is None
+                           else Variable(var, volatile=True)
+                           for var in grad_variables)
     Variable._execution_engine.run_backward(
-        tuple(variables), tuple(grad_variables), retain_variables)
+        tuple(variables), grad_variables, retain_variables)
+
+
+def differentiate(outputs, grad_outputs, inputs, only_inputs=True, retain_variables=True):
```

Comment:
This function actually works if the `outputs` are single elements but not sequence while it fails for `grad_outputs` (for the `inputs` in some cases, it breaks inside the autograd with ` One of the differentiated Variables appears to not have been used in any computation` which is very cryptic). I think it would be nice to allow having a single element here for all arguments as it is convenient, thoughts? If no, we should add checks to make sure that arguments are sequences and fail here if not.